src/widget/wsizeawarestack.cpp

```diff
@@ -70,9 +72,14 @@ class SizeAwareLayout : public QStackedLayout
     }
 };
 
-WSizeAwareStack::WSizeAwareStack(QWidget* parent)
+WSizeAwareStack::WSizeAwareStack(QWidget* parent,
+                                 ControlObject* pCurrentPageControl)
         : QWidget(parent),
           WBaseWidget(this) {
+    if (pCurrentPageControl != NULL) {
+        m_pCurrentPageControl.reset(
+                new ControlObjectSlave(pCurrentPageControl->getKey()));
+    }
```

Comment:
Why that? pCurrentPageControl is leaked. IMHO thers is no need for a ControlObjectSlave.